exporters/jaeger_thrift/src/main/java/io/opentelemetry/exporters/jaeger/JaegerThriftSpanExporter.java

```diff
@@ -0,0 +1,231 @@
+  public static final String DEFAULT_SERVICE_NAME = DEFAULT_HOST_NAME;
+
+  private static final Logger logger = Logger.getLogger(JaegerThriftSpanExporter.class.getName());
+  private static final String CLIENT_VERSION_KEY = "jaeger.version";
+  private static final String CLIENT_VERSION_VALUE = "opentelemetry-java";
+  private static final String HOSTNAME_KEY = "hostname";
+  private static final String IP_KEY = "ip";
+  private static final String IP_DEFAULT = "0.0.0.0";
+  private final ThriftSender thriftSender;
+  private final Process process;
+
+  /**
+   * Creates a new Jaeger gRPC Span Reporter with the given name, using the given channel.
+   *
+   * @param thriftSender The sender used for sending the data.
+   * @param serviceName this service's name.
+   */
+  private JaegerThriftSpanExporter(ThriftSender thriftSender, String serviceName) {
+    this.thriftSender = thriftSender;
+    String hostname;
+    String ipv4;
+
+    if (serviceName == null || serviceName.trim().length() == 0) {
+      throw new IllegalArgumentException("Service name must not be null or empty");
+    }
+
+    try {
+      hostname = InetAddress.getLocalHost().getHostName();
+      ipv4 = InetAddress.getLocalHost().getHostAddress();
+    } catch (UnknownHostException e) {
+      hostname = DEFAULT_HOST_NAME;
+      ipv4 = IP_DEFAULT;
+    }
+
+    Tag clientTag = new Tag(CLIENT_VERSION_KEY, TagType.STRING).setVStr(CLIENT_VERSION_VALUE);
+    Tag ipv4Tag = new Tag(IP_KEY, TagType.STRING).setVStr(ipv4);
+    Tag hostnameTag = new Tag(HOSTNAME_KEY, TagType.STRING).setVStr(hostname);
+
+    this.process = new Process().setServiceName(serviceName);
+    this.process.addToTags(clientTag);
+    this.process.addToTags(ipv4Tag);
+    this.process.addToTags(hostnameTag);
+  }
+
+  /**
+   * Submits all the given spans in a single batch to the Jaeger collector.
+   *
+   * @param spans the list of sampled Spans to be exported.
+   * @return the result of the operation
+   */
+  @Override
+  public CompletableResultCode export(Collection<SpanData> spans) {
+    Map<Process, List<Span>> batches =
+        spans.stream().collect(Collectors.groupingBy(SpanData::getResource)).entrySet().stream()
+            .collect(
+                Collectors.toMap(
+                    entry -> createProcess(entry.getKey()),
+                    entry -> Adapter.toJaeger(entry.getValue())));
+
+    List<CompletableResultCode> batchResults = new ArrayList<>(batches.size());
+    batches.forEach(
+        (process, jaegerSpans) -> {
+          CompletableResultCode batchResult = new CompletableResultCode();
+          try {
+            // todo: consider making truly async
+            thriftSender.send(process, jaegerSpans);
+            batchResult.succeed();
+          } catch (SenderException e) {
+            logger.log(Level.WARNING, "Failed to export spans", e);
+            batchResult.fail();
+            e.printStackTrace();
```

Comment:
Should we remove `e.printStackTrace();`?

Reply:
oh crud. yes. good catch. :)